tests/Unit/Impl/StreamxClientIntegrationTest.php

```diff
@@ -99,6 +104,59 @@ private function shouldPublishAndUnpublishDataMessage(string $key, $dataPayload)
         $this->assertDataIsUnpublished($key);
     }
 
+    //** @test */
+    public function shouldPublishAndUnpublishMultiMessageRequest() {
+        $keys = [];
+        for ($i = 0; $i < 10; $i++) {
+            $keys[] = self::MULTIMESSAGE_DATA_OBJECT_KEY . "_$i";
+        }
+
+        $this->verifyMultiMessagePublish($keys);
+        $this->verifyMultiMessageUnpublish($keys);
+    }
+
+    private function verifyMultiMessagePublish(array $keys) {
+        // given
+        $messages = [];
+        foreach ($keys as $key) {
+            $messages[] = Message::newPublishMessage($key, self::$dataArray)->build();
+        }
+
+        // when
+        $results = self::$publisher->sendMulti($messages);
+
+        // then: verify response
+        $this->assertSameSize($keys, $results);
+        for ($i = 0; $i < count($results); $i++) {
+            $result = $results[$i];
+            $this->assertInstanceOf(MessageStatus::class, $result);
+            $this->assertNotNull($result->getSuccess());
+            $this->assertEquals($keys[$i], $result->getSuccess()->getKey());
+            $this->assertIsInt($result->getSuccess()->getEventTime());
+        }
+
+        // and
+        foreach ($keys as $key) {
+            $this->assertDataIsPublished($key);
+        }
+    }
+
+    private function verifyMultiMessageUnpublish(array $keys) {
+        // given
+        $messages = [];
+        foreach ($keys as $key) {
+            $messages[] = Message::newUnpublishMessage($key)->build();
+        }
+
+        // when
+        self::$publisher->sendMulti($messages);
```

Comment:
...added [DXP-1310 Verify responses also for unpublishing](https://github.com/streamx-dev/streamx-ingestion-php/pull/6/commits/81c452855119c2bf2dcbd3845613657cee616e48) in next commit